src/map.rs

```diff
@@ -548,11 +564,13 @@ where
     ///
     /// Computes in **O(n log n + c)** time and **O(n)** space where *n* is
     /// the length of the map and *c* the capacity. The sort is stable.
-    pub fn sort_by<F>(&mut self, compare: F)
+    pub fn sort_by<F>(&mut self, mut cmp: F)
     where
         F: FnMut(&K, &V, &K, &V) -> Ordering,
     {
-        self.core.sort_by(compare)
+        self.with_entries(|entries| {
+            entries.sort_by(|a, b| cmp(&a.key, &a.value, &b.key, &b.value));
+        });
```

Comment:
style-wise I'd put `move` on both these closures so that they capture the closure by value; just something we've seen to optimize better in some cases